src/main/java/Duke.java

```diff
@@ -1,10 +1,447 @@
+     * the program.
+     * @return boolean
+     */
+    private boolean echo() {
+        Scanner user = new Scanner(System.in);
+        String input = user.nextLine();
+        if (input.equals("bye")) {
+            bye();
+            return false;
+        } else {
+            return choiceOfAction(input);
+        }
+    }
+
+    /**
+     * function adds a new Task into the taskList if valid.
+     * @param newTask
+     */
+    private void addTask(Task newTask) {
+        taskList.add(newTask);
+        System.out.println(line);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("    " + newTask);
+        System.out.format("Now you have %d tasks in the list.\n", taskList.size());
+    }
+
+    /**
+     * function removes the task located at the index of the task list.
+     * @param newInput
+     */
+    private void deleteTask(String[] newInput ) throws Exception{
+
+        if (newInput.length < 2) {
+            throw new Exception("invalid Syntax for delete instruction");
+        } else {
+            int index = Integer.parseInt(newInput[1]);
+            if (index > 0 && index <= taskList.size()) {
+                System.out.println("Noted. I've removed this task: ");
+                System.out.println("    " + taskList.get(index - 1).toString());
+                taskList.remove(index - 1);
+                System.out.format("Now you have %d tasks in the list.\n", taskList.size());
+            } else {
+                throw new Exception("index not within valid range");
+            }
+        }
+    }
+
+    /**
+     * To decide base on the input what is the next action.
+     * The decision on whether to show list, set existing task to
+     * done, or to create and record new task.
+     * @param input
+     * @return boolean
+     */
+    private boolean choiceOfAction(String input) {
+        if (input.equals("list")) {
+            showList();
+            return true;
+        } else {
+            String[] newInput = input.split(" ");
+            String instruction = newInput[0];
+            if (instruction.equals("done")) {
+                int index = Integer.parseInt(newInput[1]);
+                if (index <= taskList.size() && index > 0) {
+                    taskList.get(index - 1).done();
+                    System.out.println("Nice! I've marked this task as done: ");
+                    System.out.println("    " + this);
+                } else {
+                    System.out.println("invalid index");
+                }
+            } else if (instruction.equals("delete")) {
+                try {
+                    deleteTask(newInput);
+                }catch (Exception e) {
+                    System.out.println(e);
+                }
+            }else {
+                try {
+                    Task newTask = createTask(input);
+                    addTask(newTask);
+                } catch (TodoException tx) {
```

Comment:
I feel that seperating exception for each class is quite unneccesary. Simply a TaskException would be easier to understand.